### ISSUE TITLE
"numbered commands" broken?

### ISSUE THREAD

**Issue by DLFW** [kamiyaa/joshuto#195]
Hi,
the numbered commands seem not to work anymore. When I press a number _x_, Joshuto just says “Unmapped input: _x_”.

Is the feature broken or have I missed something? Maybe a configuration update or so?

**Comment by DLFW** [comment]
Ok, I found [this `numbered_command` command documentation](https://github.com/kamiyaa/joshuto/blob/2e8e1ccdd52d5cdc4b461b4e576c5afc9bffd2c3/docs/configuration/keymap.toml.md?plain=1#L163) in a recent commit and could add this command to my config. However, I'm a bit confused now.

1. This new command seems to be  missing in default mapping, right?
2. It seems that this command expects a default value. At least I get some `Keymap error: numbered_command: no starting character given` message otherwise. How does that make sense? A default line to go to?
3. Is it really necessary now to hit more keys to jump to a certain line? Like `ff15g` to jump to file 15? Don't get me wront, but for me this does not feel like an improvement. :smile: Maybe I'm just missing the complete picture here.

**Comment by Vinegret43** [comment]
Yep, this may be a bit inconvenient. Back then when i was implementing number-prefixed commands i just added the logic right into the main loop in `run.rs`. It checked if pressed key was a number and executed `numbered_command`, sending the pressed key to it as initial input. Looks like it was conflicting with *Alt+1, Alt+2, ...* keybindings which switch tabs, since previous logic didn't check if any modifier keys were pressed. Currently i see three possible solutions:

1. Revert the recent commit (dc5c3c96dbd4948b1be7194c2f86d473d935e199), but implement additional logic which will stop joshuto from invoking `numbered_command` if some modifier key is pressed
2. Execute `numbered_command` by pressing some key (Like `n`), but remove the need for specifying initial number, so you can just type `:numbered_command` and enter its mode without it giving you an error
3. Leave everything as is and just use something like this in your `keymap.toml`:
```
{ keys = ["1"], command = "numbered_command 1" },
{ keys = ["2"], command = "numbered_command 2" },
....
```

**Comment by kamiyaa** [comment]
Yes, you guys are correct. I've decided to move this logic into the `numbered_command` to prevent conflicts with other commands if a user wants to map numbers to other commands.

> 3. Leave everything as is and just use something like this in your keymap.toml:
```
{ keys = ["1"], command = "numbered_command 1" },
{ keys = ["2"], command = "numbered_command 2" },
...
```
Yes, this is the solution I had in mind and I can add it to the default keymap if you guys want.

=======

> 3. Is it really necessary now to hit more keys to jump to a certain line? Like ff15g to jump to file 15? Don't get me wront, but for me this does not feel like an improvement. smile Maybe I'm just missing the complete picture here.

Correct me if I'm wrong, but I don't think this will be necessary as stated by @Vinegret43 
This shouldn't cause any feature regressions, but just a bit more explicit with the keymaps.

> 3. Leave everything as is and just use something like this in your keymap.toml:
```
{ keys = ["1"], command = "numbered_command 1" },
{ keys = ["2"], command = "numbered_command 2" },
```

=======

> 2. It seems that this command expects a default value. At least I get some Keymap error: numbered_command: no starting character given message otherwise. How does that make sense? A default line to go to?

Yep, since the initial implementation was triggered from a keystroke (ie. a number), like what @Vinegret43 (3) said, just need to pass in the keystroke. This can potentially be more customizable because you can map a non-number to a number input.
ie. `{ keys = ["f1"], command = "numbered_command 1" },`

Let me know what you guys think!

**Comment by Vinegret43** [comment]
> and I can add it to the default keymap if you guys want

That would be quite nice, since this is what you want to have out-of-the-box when running Joshuto, and some new users may not be able to figure this out

**Comment by kamiyaa** [comment]
closed via 30111d20afc16c31a74afb7cf134f4ad8f1ade18
